app/service/planning_svc.py

```diff
@@ -47,9 +43,10 @@ def __init__(self):
         """
         link_status = await self._default_link_status(operation)
         links = []
-        if agent and await self._check_untrusted_agents_allowed(agent=agent, operation=operation,
+        if agent:
+            if await self._check_untrusted_agents_allowed(agent=agent, operation=operation,
```

Comment:
Isn't this functionally the same as the original code block?

Reply:
and the code blocks are identical:
```
if await self._check_untrusted_agents_allowed(agent=agent, operation=operation,
                                                                msg='no cleanup-link created'):
                links.extend(
                    await self._generate_cleanup_links(operation=operation, agent=agent, link_status=link_status)
                    )
```

Make them a private function

Reply:
`def _append_cleanup_links() might make sense...`

Reply:
No to first question. Execution will go into block if agent is present. But then if agent doesnt pass untrusted checks, then no links are created, function exits. Versus before,  would proceed to 'else' block if that was the case, and then (possibly) return links for other agents.

Reply:
Ah makes sense.  If you address the second item, it's irrelevant to the first anyways 👍